#!/usr/bin/env python

PACKAGE = "meka_velocity_smoother"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("speed_lim_v", double_t, 0, "Maximum linear velocity", 1.0, 0.0, 5.0)
gen.add("speed_lim_w", double_t, 0, "Maximum angular velocity", 2.0, 0.0, 10.0)

gen.add("accel_lim_v", double_t, 0, "Maximum linear acceleration", 10.0, 0.0, 40.0)
gen.add("accel_lim_w", double_t, 0, "Maximum angular acceleration", 20.0, 0.0, 80.0)

gen.add("jerk_lim_v", double_t, 0, "Maximum linear jerk", 10.0, 0.0, 40.0)
gen.add("jerk_lim_w", double_t, 0, "Maximum angular jerk", 20.0, 0.0, 80.0)

gen.add("decel_factor", double_t, 0, "Factor to speed up stopping", 4.0, 0.0, 10.0)

# Second arg is node name it will run in (doc purposes only), third is generated filename prefix
exit(gen.generate(PACKAGE, "velocity_smoother_configure", "params"))
